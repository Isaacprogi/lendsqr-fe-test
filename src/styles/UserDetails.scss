@import '../styles/variables/mixins';
@import '../styles/variables/mixins';


.loading{
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: blue;
}


.user-details-board{
    height: 100%;
    max-width: 100%;
    overflow-x:hidden;


    .header {
        display: flex;
        align-items: flex-start;
        justify-content: space-between;
        flex-direction: column;
        width: 100%;
        padding-left: 1rem;
        padding-right: 1rem;


        .right-side {
            display: flex;
            align-items: flex-start;
            justify-content: flex-start;
            flex-direction: column;
            width: 100%;

            >span {
                margin-bottom: 1rem;
                margin-top: 1rem;
                color: $textPrimary;
                min-width: 8rem;
                cursor: pointer;

                img {
                    margin-right: .5rem;
                }
            }

            :nth-last-child(1) {
                font-weight: 500;
                font-size: 24px;
                color: $textPrimary;
            }




        }

        .left-side {
            display: flex;
            align-items: center;
            justify-content: space-between;
            width: 100%;
            margin-top: 1rem;

            >span {
                margin-right: 1rem;
                color: #E4033B;
                overflow: hidden;

                &:hover {
                    background-color: #e3f3f3;
                    color: white;
                    border: none;
                }

            }

            :nth-last-child(1) {
                color: #39CDCC;
                overflow: hidden;

                &:hover {
                    background-color: #e3f3f3;
                    color: white;
                    border: none;
                }
            }

            span {
                width: 170px;
                height: 40px;
                display: flex;
                align-items: center;
                justify-content: center;
                text-align: center;
                border: 1px solid #E4033B;
                border-radius: 8px;
                cursor: pointer;
                font-weight: 600;
                font-size: 14px;

                @include mediumDevices {
                    width: 150px;
                }
            }



        }

        @include smallDevices {
            flex-direction: row;

        }
    }

    .header-container {
        margin-top: 1rem;
        border: 1px solid rgba(33, 63, 125, 0.06);
        box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
        border-radius: 4px;
        overflow: hidden;
        width: 100%;
        display: flex;
        flex-direction: column;


        .header-data {
            display: flex;
            flex-direction: column;
            justify-content: center;
            align-items: center;
            width: 100%;
            overflow: hidden;

            padding: 1rem;


            .avatar {
                background: #e9d5ec;
                min-width: 90px;
                max-width: 90px;
                height: 90px;
                border-radius: 50%;
                display: flex;
                align-items: center;
                justify-content: center;
                overflow: hidden;

                margin-bottom: 1rem;

                @include smallDevices {
                    margin-right: 1rem;
                    margin-bottom: 0;

                }


            }

            .details-1 {
                display: flex;
                flex-direction: column;
                align-items: center;
                justify-content: center;


                height: 100%;
                word-break: break-all;
                margin-bottom: 1rem;
        

                @include smallDevices {
                    padding-right: 1rem;
                    border-right: 1px solid #d3d5d9;
                    margin-bottom: 0;
                    width: max-content;
                    margin-right: 1rem;
                }
            }

            .details-2 {
                min-width: max-content;
                display: flex;
                flex-direction: column;

                word-break: break-all;

                align-items: center;
                justify-content: center;


                height: 100%;
                margin-bottom: 1rem;

                @include smallDevices {
                    padding-right: 1rem;
                    border-right: 1px solid #d3d5d9;
                    margin-bottom: 0;
                    align-items: flex-start;
                    margin-right: 1rem;
                    width: max-content;
                }


            }

            .details-3 {
                display: flex;
                flex-direction: column;
                overflow: hidden;
               
                justify-content: center;
                align-items: center;
                width: max-content;

                

                @include smallDevices {
                    width: max-content;
                    align-items: flex-start;
                }
            }


            @include smallDevices {
                flex-direction: row;
                align-items: center;
                justify-content: flex-start;
                min-height: 8rem;

            }

        }

        .header-options {
            display: flex;
            align-items: center;
            justify-content: space-between;
            overflow-y: auto;
            width: 100%;
            padding-left: 1rem;
            padding-right: 1rem;

            &::-webkit-scrollbar {
                display: none;
            }


            span {
                min-width: max-content;
                cursor: pointer;
                font-weight: 700;
                margin-right: 1rem;
                cursor: pointer;
            }

            span.active {
                border-bottom: 1px solid rgb(173, 166, 166);
            }

            :last-child {
                margin-right: 0;
            }
        }

    }

    .user-body-data {
        display: flex;
        flex-direction: column;
        height: max-content;
        width: 100%;
        background: $whiteColor;
        border: 1px solid rgba(33, 63, 125, 0.06);
        box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
        border-radius: 4px;
        margin-top: 1rem;
        padding: 1rem;

        .sec {
            display: flex;
            flex-direction: column;
            width: 100%;
            border-bottom: .5px solid rgb(230, 221, 221);
            padding-bottom: 1rem;

            >span {
                min-height: 4rem;
            }

            .sec-header {
                margin-bottom: 1rem;
                margin-top: 1rem;
                font-weight: 600;
            }

            .sec-data {
                width: 100%;
                display: flex;
                flex-wrap: wrap;
                row-gap: 1.5rem;






                .details {
                    display: flex;
                    flex-direction: column;
                    min-width: 10rem;
                    overflow: hidden;
                    margin-right: 1rem;
                    word-break: break-all;

                    @include smallerDevices {
                        max-width: 10rem;
                    }

                    .data-title {
                        font-size: 14px;
                        color: $textSecondary;
                        margin-bottom: .3rem;
                        word-wrap: break-word;
                        font-weight: 600;
                    }

                    .data-value {
                        font-weight: 400;
                        font-size: 12px;
                        line-height: 19px;
                        color: $textSecondary;
                    }


                }
            }
        }

        .sec.last{
            border-bottom: none;
        }

        .second-sec {
            background-color: blue;
        }
    }

}