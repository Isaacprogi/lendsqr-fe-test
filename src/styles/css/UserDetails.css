.loading {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  background-color: blue;
}

.user-details-board {
  height: 100%;
  max-width: 100%;
  overflow-x: hidden;
}
.user-details-board .header {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: start;
      -ms-flex-align: start;
          align-items: flex-start;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  width: 100%;
  padding-left: 1rem;
  padding-right: 1rem;
}
.user-details-board .header .right-side {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: start;
      -ms-flex-align: start;
          align-items: flex-start;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  width: 100%;
}
.user-details-board .header .right-side > span {
  margin-bottom: 1rem;
  margin-top: 1rem;
  color: #213F7D;
  min-width: 8rem;
  cursor: pointer;
}
.user-details-board .header .right-side > span img {
  margin-right: 0.5rem;
}
.user-details-board .header .right-side :nth-last-child(1) {
  font-weight: 500;
  font-size: 24px;
  color: #213F7D;
}
.user-details-board .header .left-side {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  width: 100%;
  margin-top: 1rem;
}
.user-details-board .header .left-side > span {
  margin-right: 1rem;
  color: #E4033B;
  overflow: hidden;
}
.user-details-board .header .left-side > span:hover {
  background-color: #e3f3f3;
  color: white;
  border: none;
}
.user-details-board .header .left-side :nth-last-child(1) {
  color: #39CDCC;
  overflow: hidden;
}
.user-details-board .header .left-side :nth-last-child(1):hover {
  background-color: #e3f3f3;
  color: white;
  border: none;
}
.user-details-board .header .left-side span {
  width: 170px;
  height: 40px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  text-align: center;
  border: 1px solid #E4033B;
  border-radius: 8px;
  cursor: pointer;
  font-weight: 600;
  font-size: 14px;
}
@media screen and (min-width: 768px) {
  .user-details-board .header .left-side span {
    width: 150px;
  }
}
@media screen and (min-width: 568px) {
  .user-details-board .header {
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
        -ms-flex-direction: row;
            flex-direction: row;
  }
}
.user-details-board .header-container {
  margin-top: 1rem;
  border: 1px solid rgba(33, 63, 125, 0.06);
  -webkit-box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
          box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
  border-radius: 4px;
  overflow: hidden;
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}
.user-details-board .header-container .header-data {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: 100%;
  overflow: hidden;
  padding: 1rem;
}
.user-details-board .header-container .header-data .avatar {
  background: #e9d5ec;
  min-width: 90px;
  max-width: 90px;
  height: 90px;
  border-radius: 50%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  overflow: hidden;
  margin-bottom: 1rem;
}
@media screen and (min-width: 568px) {
  .user-details-board .header-container .header-data .avatar {
    margin-right: 1rem;
    margin-bottom: 0;
  }
}
.user-details-board .header-container .header-data .details-1 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  height: 100%;
  word-break: break-all;
  margin-bottom: 1rem;
}
@media screen and (min-width: 568px) {
  .user-details-board .header-container .header-data .details-1 {
    padding-right: 1rem;
    border-right: 1px solid #d3d5d9;
    margin-bottom: 0;
    width: -webkit-max-content;
    width: -moz-max-content;
    width: max-content;
    margin-right: 1rem;
  }
}
.user-details-board .header-container .header-data .details-2 {
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  word-break: break-all;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  height: 100%;
  margin-bottom: 1rem;
}
@media screen and (min-width: 568px) {
  .user-details-board .header-container .header-data .details-2 {
    padding-right: 1rem;
    border-right: 1px solid #d3d5d9;
    margin-bottom: 0;
    -webkit-box-align: start;
        -ms-flex-align: start;
            align-items: flex-start;
    margin-right: 1rem;
    width: -webkit-max-content;
    width: -moz-max-content;
    width: max-content;
  }
}
.user-details-board .header-container .header-data .details-3 {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  overflow: hidden;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}
@media screen and (min-width: 568px) {
  .user-details-board .header-container .header-data .details-3 {
    width: -webkit-max-content;
    width: -moz-max-content;
    width: max-content;
    -webkit-box-align: start;
        -ms-flex-align: start;
            align-items: flex-start;
  }
}
@media screen and (min-width: 568px) {
  .user-details-board .header-container .header-data {
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
        -ms-flex-direction: row;
            flex-direction: row;
    -webkit-box-align: center;
        -ms-flex-align: center;
            align-items: center;
    -webkit-box-pack: start;
        -ms-flex-pack: start;
            justify-content: flex-start;
    min-height: 8rem;
  }
}
.user-details-board .header-container .header-options {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: justify;
      -ms-flex-pack: justify;
          justify-content: space-between;
  overflow-y: auto;
  width: 100%;
  padding-left: 1rem;
  padding-right: 1rem;
}
.user-details-board .header-container .header-options::-webkit-scrollbar {
  display: none;
}
.user-details-board .header-container .header-options span {
  min-width: -webkit-max-content;
  min-width: -moz-max-content;
  min-width: max-content;
  cursor: pointer;
  font-weight: 700;
  margin-right: 1rem;
  cursor: pointer;
}
.user-details-board .header-container .header-options span.active {
  border-bottom: 1px solid rgb(173, 166, 166);
}
.user-details-board .header-container .header-options :last-child {
  margin-right: 0;
}
.user-details-board .user-body-data {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  width: 100%;
  background: white;
  border: 1px solid rgba(33, 63, 125, 0.06);
  -webkit-box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
          box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
  border-radius: 4px;
  margin-top: 1rem;
  padding: 1rem;
}
.user-details-board .user-body-data .sec {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  width: 100%;
  border-bottom: 0.5px solid rgb(230, 221, 221);
  padding-bottom: 1rem;
}
.user-details-board .user-body-data .sec > span {
  min-height: 4rem;
}
.user-details-board .user-body-data .sec .sec-header {
  margin-bottom: 1rem;
  margin-top: 1rem;
  font-weight: 600;
}
.user-details-board .user-body-data .sec .sec-data {
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
      flex-wrap: wrap;
  row-gap: 1.5rem;
}
.user-details-board .user-body-data .sec .sec-data .details {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  min-width: 10rem;
  overflow: hidden;
  margin-right: 1rem;
  word-break: break-all;
}
@media screen and (min-width: 450px) {
  .user-details-board .user-body-data .sec .sec-data .details {
    max-width: 10rem;
  }
}
.user-details-board .user-body-data .sec .sec-data .details .data-title {
  font-size: 14px;
  color: #545F7D;
  margin-bottom: 0.3rem;
  word-wrap: break-word;
  font-weight: 600;
}
.user-details-board .user-body-data .sec .sec-data .details .data-value {
  font-weight: 400;
  font-size: 12px;
  line-height: 19px;
  color: #545F7D;
}
.user-details-board .user-body-data .sec.last {
  border-bottom: none;
}
.user-details-board .user-body-data .second-sec {
  background-color: blue;
}/*# sourceMappingURL=UserDetails.css.map */