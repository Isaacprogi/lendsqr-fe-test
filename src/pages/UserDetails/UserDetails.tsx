import { useState } from 'react'
import UserProfile from '../../assets/svg/user-activities/UserProfile.svg'
import ArrowBack from '../../assets/svg/user-activities/ArrowBack.svg'
import StarFilled from '../../assets/svg/user-activities/StarFilled.svg'
import StarOutline from '../../assets/svg/user-activities/StarOutline.svg'
import { UserDataInfo } from '../../components/UserDataInfo/UserDataInfo'
import { useParams } from "react-router-dom"
import '../../styles/css/UserDetails.css'
import { useNavigate } from 'react-router-dom'
import { UserDetailsGetApiData } from "../../utils/types/interface_and_types"
import { useEffect } from "react"
import axios from 'axios'

export const UserDetails = () => {
    const { id } = useParams()
    const [userDetails, setUserDetails] = useState<UserDetailsGetApiData>()
    const [headerActive, setHeaderActive] = useState<String>('')
    const [loading, setLoading] = useState<boolean>(false)

    const navigate = useNavigate()




    useEffect(() => {
        const getUserDetails = async () => {
            setLoading(true)
            try {
                const response = await axios.get(`https://6270020422c706a0ae70b72c.mockapi.io/lendsqr/api/v1/users/${id}`)
                setUserDetails(response?.data)
                setLoading(false)
            } catch (error) {
                setLoading(false)
                console.log(error)
            }
        }
        getUserDetails()
    }, [id])


    const handleHeaderActive = (value: string) => {
        setHeaderActive(value)
    }




    return loading === false && userDetails ? (
        <div className='user-details-board'>

            <div className="header">
                <div className="right-side" >
                    <span ><img onClick={() => navigate(-1)} src={ArrowBack} alt="navigate-back-icon" />back to users</span>
                    <span data-testid="user-details">User Details</span>
                </div>

                <div className="left-side">
                    <span data-testid='black-list-user'>BLACKLIST USER</span>
                    <span data-testid='activate-user'>ACTIVATE USER</span>
                </div>
            </div>

            <div className="header-container">
                <section className="header-data">
                    <div data-testid="avatar" className="avatar">
                        {
                            userDetails?.profile.avatar ?
                                <img src={userDetails.profile.avatar} alt="avatar-logo" />
                                : <img src={UserProfile} alt="avatar-profile-pic" />
                        }
                    </div>
                    <div className="details-1">
                        {
                            userDetails && <>
                                <span className="user-details-name">{userDetails?.profile.firstName + " " + userDetails?.profile.lastName}</span>
                                <span className="user-details-number">LSGF587G90</span>
                            </>
                        }
                    </div>
                    <div className="details-2">
                        <span data-testid="user-tier">User's Tier</span>
                        <span>
                            <img src={StarFilled} alt="star-icon" />
                            <img src={StarOutline} alt="star-icon" />
                            <img src={StarOutline} alt="star-icon" />
                        </span>
                    </div>


                    <div className="details-3">
                        {userDetails && <>
                            <span data-testid="user-amount">{userDetails?.accountBalance}</span>
                            <span className="bank-details" data-testid="user-bank">{`${userDetails?.accountNumber}/Providus Bank`}</span>
                        </>}
                    </div>

                </section>
                <section className={`header-options `}>
                    <span data-testid="header-options-general" className={`${headerActive === 'General' ? 'active' : ""}`} onClick={() => handleHeaderActive('General')} >General Setting</span>
                    <span data-testid="header-options-documents" className={`${headerActive === 'Documents' ? 'active' : ""}`} onClick={() => handleHeaderActive('Documents')} >Documents</span>
                    <span data-testid="header-options-bank" className={`${headerActive === 'Bank' ? 'active' : ""}`} onClick={() => handleHeaderActive('Bank')} >Bank Details</span>
                    <span data-testid="header-options-loans" className={`${headerActive === 'Loans' ? 'active' : ""}`} onClick={() => handleHeaderActive('Loans')} >Loans</span>
                    <span data-testid="header-options-savings" className={`${headerActive === 'Savings' ? 'active' : ""}`} onClick={() => handleHeaderActive('Savings')} >Savings</span>
                    <span data-testid="header-options-app" className={`${headerActive === 'App' ? 'active' : ""}`} onClick={() => handleHeaderActive('App')} >App and System</span>
                </section>
            </div>

            <div className="user-body-data">
                <section className="sec">
                    <span data-testid='sec-header' className="sec-header">
                        Personal Info
                    </span>

                    <span>
                        {
                            userDetails && <div className="sec-data">
                                <UserDataInfo dataIdValue="personal-info-value" dataIdTitle="personal-info-title" dataTitle="FULL NAME" dataValue={userDetails?.profile.firstName + " " + userDetails?.profile.lastName} />
                                <UserDataInfo dataIdValue="personal-info-value" dataIdTitle="personal-info-title" dataTitle="PHONE NUMBER" dataValue={userDetails?.profile.phoneNumber} />
                                <UserDataInfo dataIdValue="personal-info-value" dataIdTitle="personal-info-title" dataTitle="EMAIL ADDRESS" dataValue={userDetails?.email} />
                                <UserDataInfo dataIdValue="personal-info-value" dataIdTitle="personal-info-title" dataTitle="BVN" dataValue={userDetails?.profile.bvn} />
                                <UserDataInfo dataIdValue="personal-info-value" dataIdTitle="personal-info-title" dataTitle="GENDER" dataValue={userDetails.profile.gender} />
                                <UserDataInfo dataIdValue="personal-info-value" dataIdTitle="personal-info-title" dataTitle="MARITAL STATUS" dataValue={userDetails.profile.gender} />
                                <UserDataInfo dataIdValue="personal-info-value" dataIdTitle="personal-info-title" dataTitle="CHILDREN" dataValue="none" />
                                <UserDataInfo dataIdValue="personal-info-value" dataIdTitle="personal-info-title" dataTitle="TYPE OF RESIDENCE" dataValue="parents appartment" />
                            </div>
                        }
                    </span>
                </section>

                <section className="sec">
                    <span data-testid='sec-header' className="sec-header">
                        Entertainment And Employment
                    </span>

                    <span>
                        {
                            userDetails && <div className="sec-data">
                                <UserDataInfo dataIdTitle="entertainment-title" dataIdValue='entertainment-value' dataTitle="LEVEL OF EDUCATION" dataValue={userDetails.education.level} />
                                <UserDataInfo dataIdTitle="entertainment-title" dataIdValue='entertainment-value' dataTitle="EMPLOYMENT STATUS" dataValue={userDetails.education.employmentStatus} />
                                <UserDataInfo dataIdTitle="entertainment-title" dataIdValue='entertainment-value' dataTitle="SECTOR OF EMPLOYMENT" dataValue={userDetails.education.sector} />
                                <UserDataInfo dataIdTitle="entertainment-title" dataIdValue='entertainment-value' dataTitle="DURATION OF EMPLOYMENT" dataValue={userDetails.education.duration} />
                                <UserDataInfo dataIdTitle="entertainment-title" dataIdValue='entertainment-value' dataTitle="OFFICE EMAIL" dataValue={userDetails.education.officeEmail} />
                                <UserDataInfo dataIdTitle="entertainment-title" dataIdValue='entertainment-value' dataTitle="MONTHLY INCOME" dataValue={userDetails.education.monthlyIncome[1] + "-" + userDetails.education.monthlyIncome[0]} />
                                <UserDataInfo dataIdTitle="entertainment-title" dataIdValue='entertainment-value' dataTitle="LOAN PAYMENT" dataValue={userDetails.education.loanRepayment} />
                            </div>
                        }
                    </span>
                </section>

                <section className="sec">
                    <span data-testid='sec-header' className="sec-header">
                        Socials
                    </span>
                    <span>

                        {
                            userDetails && <div className="sec-data">
                                <UserDataInfo dataIdValue="socials-value" dataIdTitle="socials-title" dataTitle="TWITTER" dataValue={userDetails?.socials.twitter} />
                                <UserDataInfo dataIdValue="socials-value" dataIdTitle="socials-title" dataTitle="FACEBOOK" dataValue={userDetails.socials.facebook} />
                                <UserDataInfo dataIdValue="socials-value" dataIdTitle="socials-title" dataTitle="INSTAGRAM" dataValue={userDetails?.socials.instagram} />

                            </div>
                        }
                    </span>
                </section>

                <section className="sec last">
                    <span data-testid='sec-header' className="sec-header">
                        Garantors
                    </span>

                    <span>
                        {
                            userDetails && <div className="sec-data">
                                <UserDataInfo dataIdValue="guarantors-value" dataIdTitle="guarantors-title" dataTitle="FULL NAME" dataValue={userDetails.guarantor.firstName + " " + userDetails.guarantor.lastName} />
                                <UserDataInfo dataIdValue="guarantors-value" dataIdTitle="guarantors-title" dataTitle="PHONE NUMBER" dataValue={userDetails.guarantor.phoneNumber} />
                                <UserDataInfo dataIdValue="guarantors-value" dataIdTitle="guarantors-title" dataTitle="EMAIL ADDRESS" dataValue="debby@gmail.com" />
                                <UserDataInfo dataIdValue="guarantors-value" dataIdTitle="guarantors-title" dataTitle="RELATIONSHIP" dataValue="sister" />


                            </div>
                        }
                    </span>

                </section>



            </div>

        </div>
    ) : <div className="skeleton">
        <div className="skeleton-line-1"></div>
        <div className="skeleton-line-2"></div>
        <div className="skeleton-line-3"></div>
    </div>
}
