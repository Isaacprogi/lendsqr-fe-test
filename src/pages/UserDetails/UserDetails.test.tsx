import { UserDetails } from './UserDetails';
import { render, waitFor, screen, act } from '@testing-library/react';
import axios from 'axios';
import { MemoryRouter, BrowserRouter, Routes, Route } from 'react-router-dom';
import { userDetailsMockData } from '../../utils/data/mockdata';
import { vi } from 'vitest';
import renderer from 'react-test-renderer';

vi.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe('UserDetails', () => {

  test('data should be present after API call is made', async () => {
    mockedAxios.get.mockResolvedValueOnce({ data: userDetailsMockData });

    await act(async () => {
      render(
        <MemoryRouter initialEntries={[`/user-details/1`]}>
          <Routes>
            <Route path='/user-details/:id' element={<UserDetails />} />
          </Routes>
        </MemoryRouter>
      );
    });

    expect(axios.get).toHaveBeenCalledWith(`https://6270020422c706a0ae70b72c.mockapi.io/lendsqr/api/v1/users/1`);
    expect(axios.get).toHaveBeenCalledTimes(1);

    await waitFor(() => {
      const divElement = screen.getByTestId('user-amount');
      console.log(divElement.innerHTML); // Debugging: log the innerHTML of the div element
      expect(divElement).toBeInTheDocument();
      expect(divElement.innerHTML).toBe(userDetailsMockData.accountBalance);
    });
  });

  test('should match snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <UserDetails />
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
