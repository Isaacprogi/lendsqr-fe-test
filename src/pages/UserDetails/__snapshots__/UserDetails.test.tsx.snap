// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserDetails > should match snapshot 1`] = `
<div
  className="skeleton"
>
  <div
    className="skeleton-line-1"
  />
  <div
    className="skeleton-line-2"
  />
  <div
    className="skeleton-line-3"
  />
</div>
`;
